src/client/api/event.c

```diff
@@ -1243,12 +1244,29 @@ daos_event_priv_wait()
 
 	/* Wait on the event to complete */
 	while (evx->evx_status != DAOS_EVS_READY) {
+		int rc2;
+
 		rc = crt_progress_cond(evx->evx_ctx, 0, ev_progress_cb, &epa);
-		if (rc == 0)
+
+		/** progress succeeded, loop can exit if event completed */
+		if (rc == 0) {
 			rc = ev_thpriv.ev_error;
```

Comment:
"rc" will be overwritten by the next crt_progress_cond() immediately, we might not need to save it?
The main behavior changed by this patch is, we originally exit from the loop for ev_thpriv.ev_error != 0, now we don't do that anymore, is this intended?

Reply:
i thought that if the error is set, then the loop will exit anyway in the check?
maybe im wrong..

Reply:
i think you're right though, we should exit the loop. i will fix this first on master.

Reply:
created this PR.
https://github.com/daos-stack/daos/pull/4981
will integrate into this 1.2 pr once that lands